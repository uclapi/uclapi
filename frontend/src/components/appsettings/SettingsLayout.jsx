import axios from 'axios'
import PropTypes from 'prop-types'
import React from 'react'

// Components
import { ButtonView, CardView, Column, Row, TextView } from 'Layout/Items.jsx'

const styles = {
  noPadding: {
    marginBottom: 0,
    marginLeft: 0,
  },
  title: {
    marginLeft: 0,
    margin: `20px 0`,
  },
}

class SettingsLayout extends React.Component {
  static propTypes = {
    fullname: PropTypes.string,
    department: PropTypes.string,
    authorised_apps: PropTypes.array,
  }

  constructor(props) {
    super(props)
    const { authorised_apps } = this.props
    this.state = {
      isDeleted: authorised_apps.map(() => false),
    }
  }

  render() {
    const { authorised_apps, fullname, department } = this.props
    const { isDeleted } = this.state

    const appCount = authorised_apps.reduce(
      (prev, curApp) => curApp.active ? (prev + 1) : prev,
      0
    )
    const isApps = appCount === 0

    return (
      <Row
        height='fit-content'
        styling='splash-parallax'
        style={{ minHeight: `100vh` }}
      >
        <Column
          width='2-3'
          horizontalAlignment='center'
          style={{ marginTop: `50px` }}
        >

          <TextView
            text={`Account`}
            heading={2}
            align={`left`}
            style={styles.title}
          />

          <CardView width='1-1' type='default' noPadding>
            <Row noPadding>
              <Column width='1-1' horizontalAlignment='center' >
                <CardView
                  width='1-2'
                  type="transparent"
                  noShadow
                >
                  <TextView
                    text={fullname}
                    heading={2}
                    align={`left`}
                    style={styles.noPadding}
                  />
                  <TextView
                    text={department}
                    heading={4}
                    align={`left`}
                    style={styles.noPadding}
                  />
                </CardView>
                <CardView
                  width='1-2'
                  type="transparent"
                  noShadow
                >
                  <ButtonView
                    type="alternate"
                    link={`\logout`}
                    text={`Logout`}
                    style={{
                      float: `right`,
                      cursor: `pointer`,
                    }}
                  />
                </CardView>
              </Column>
            </Row>
          </CardView>

          <TextView
            text={`Permissions`}
            heading={2}
            align={`left`}
            style={styles.title}
          />

          <CardView width='1-1' type='default' noPadding>
            <Row noPadding>
              <Column width='1-1'
                horizontalAlignment='center'
                style={{
                  paddingTop: 20,
                  paddingBottom: 20,
                }}
              >
                {isApps ? (
                  <>
                    <TextView
                      text={`No authorised apps yet. Try UCL Assistant?`}
                      heading={2}
                      align={`center`}
                      style={styles.noPadding}
                    />
                    <ButtonView text="Download"
                      type="alternate"
                      link={`/marketplace/uclassistant`}
                      style={{
                        marginBottom: 0,
                        cursor: `pointer`,
                      }}
                    />
                  </>
                ) : (
                    <>
                      {authorised_apps.map((app, i) =>
                        <>
                          {app.active && !isDeleted[i] ? (
                            <>
                              <div className="default tablet">
                                <CardView
                                  width='1-2'
                                  type="transparent"
                                  noShadow
                                >
                                  <TextView text={app.app.name.toUpperCase()}
                                    heading={2}
                                    align={`left`}
                                    style={styles.noPadding}
                                  />
                                  <TextView text={app.app.creator.name}
                                    heading={4}
                                    align={`left`}
                                    style={styles.noPadding}
                                  />
                                </CardView>
                                <CardView
                                  width='1-2'
                                  type="transparent"
                                  noShadow
                                >
                                  <ButtonView
                                    type="alternate"
                                    onClick={
                                      this.handleChange(app.app.client_id, i)
                                    }
                                    text={`Revoke Permissions`}
                                    style={{
                                      float: `right`,
                                      cursor: `pointer`,
                                    }}
                                  />
                                </CardView>
                              </div>
                              <div className="mobile">
                                <CardView
                                  width='1-1'
                                  type="transparent"
                                  noShadow
                                >
                                  <TextView text={app.app.name.toUpperCase()}
                                    heading={2}
                                    align={`center`}
                                    style={styles.noPadding}
                                  />
                                  <TextView text={app.app.creator.name}
                                    heading={4}
                                    align={`center`}
                                    style={styles.noPadding}
                                  />
                                </CardView>
                                <CardView
                                  width='1-1'
                                  type="transparent"
                                  noShadow
                                >
                                  <ButtonView
                                    type="alternate"
                                    onClick={
                                      this.handleChange(app.app.client_id, i)
                                    }
                                    text={`Revoke Permissions`}
                                    style={{ cursor: `pointer` }}
                                  />
                                </CardView>
                              </div>
                            </>
                          ) : null}
                        </>
                      )}
                    </>
                  )}
              </Column>
            </Row>
          </CardView>
        </Column>
      </Row>
    )
  }

  handleChange = (client_id, index) => () => {
    const { authorised_apps } = this.props

    // Add toast to confirm deauthorisation
    const deauthoriseConfirmation = `Are you sure you want to deauthorise `
      + authorised_apps[index].app.name + `? It will probably stop working and `
      + `you'll need to re-authenticate with it again if you want to use it`
    if (confirm(deauthoriseConfirmation)) {
      this.deauthoriseApp(client_id, index)
    }
  }

  logout = () => {
    if (confirm("Are you sure you want to logout")) {
      let path = `/logout`;
      let history = useHistory();
      history.push(path);
    }
  }

  deauthoriseApp = (client_id, index) => {
    // Call function in back end to delete scope
    axios.get(`/oauth/deauthorise`, {
      params: {
        client_id: client_id,
      },
      xsrfHeaderName: `X-CSRFToken`,
    }).then(response => {
      // Log success in console
      console.log(`Successfully de-authorised app: `)
      console.log(response)

      const { isDeleted } = this.state
      isDeleted[index] = true

      // Update react to no longer render this app
      this.setState({
        isDeleted: isDeleted,
      })
    }).catch(error => {
      // Handle error
      console.log(error)
    })
  }
}

export default SettingsLayout
