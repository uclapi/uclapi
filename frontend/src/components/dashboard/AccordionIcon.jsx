import React from 'react'
import { FiChevronDown, FiChevronRight } from 'react-icons/fi'
import './AccordionIcon.scss'

const AccordionIcon = ({ isActive }) => (
  <div className="accordion-icon">
    {isActive ? <FiChevronDown /> : <FiChevronRight />}
  </div>
)

export default AccordionIcon