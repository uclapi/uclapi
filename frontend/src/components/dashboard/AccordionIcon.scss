.accordion-icon {
  margin-right: 0.5rem;
  display: inline-block;

  > svg {
    vertical-align: middle;
  }
}
