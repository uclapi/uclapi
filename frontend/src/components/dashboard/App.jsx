import React from 'react'

// Stylings
import { styles } from 'Layout/data/dashboard_styles.jsx'

// Components
import { CardView, Column, Footer, NavBar, Row, TextView, 
  ButtonView, Field, ConfirmBox, CheckBox, Container } from 'Layout/Items.jsx'
import { editIcon, cancelIcon } from 'Layout/Icons.jsx'

// External dependencies
import Collapse, { Panel } from 'rc-collapse'

/**
	UI Definition of the app, all of the "actions" should be passed in from
	the dashboard page and they will do all the heavy lifting whereas this
	page is only a ui representation
**/

const arrowPath = 'M869 487.8L491.2 159.9c-2.9-2.5-6.6-3.9-10.5-3.9h-88' +
  '.5c-7.4 0-10.8 9.2-5.2 14l350.2 304H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.' +
  '6 8 8 8h585.1L386.9 854c-5.6 4.9-2.2 14 5.2 14h91.5c1.9 0 3.8-0.7 5.' +
  '2-2L869 536.2c14.7-12.8 14.7-35.6 0-48.4z';

function expandIcon({ isActive }) {
  return (
    <i style={{ marginRight: '.5rem' }}>
      <svg
        viewBox="0 0 1024 1024"
        width="1em"
        height="1em"
        fill="currentColor"
        style={{
          verticalAlign: '-.125em',
          transition: 'transform .2s',
          transform: `rotate(${isActive ? 90 : 0}deg)`,
        }}
      >
        <path d={arrowPath} p-id="5827"></path>
      </svg>
    </i>
  );
}

export default class App extends React.Component {
	
	constructor(props) { 
		super(props)

		this.DEBUGGING = true

		const { app } = this.props

		const updated = this.timeSince(new Date(app.updated))
        const created = this.timeSince(new Date(app.created))

        this.state = {
        	updated: updated,
        	created: created
        }
	}

	componentDidUpdate(prevProps) {
		const { app, index } = this.props

		if(this.DEBUGGING) { console.log("App updated") }

		if(app != prevProps.app || index != prevProps.index) { 
			const { app } = this.props

			const updated = this.timeSince(new Date(app.updated))
	        const created = this.timeSince(new Date(app.created))

	        this.setState({
	        	updated: updated,
	        	created: created
	        })
		}
	}

	render() {
		const { updated, created } = this.state
		const { app, actions, index } = this.props

		if(this.DEBUGGING) { console.log("re-rendering app, name: " + app.name) }

		const trashColor = "red"

		console.log(app.oauth.scopes)

		return (
		<Collapse expandIcon={expandIcon}>
        <Panel header={app.name} showArrow>

			<Container styling='transparent' noPadding>
				<Row width="1-1">
					<Column 
						width="1-1" 
						className="title-holder"
					>
						<Field
						title="Title: "
						content={app.name}
						onSave={(value) => { actions.saveEditTitle(index, value) }}
						isSmall={true}
						/>
						<Field
							title="API Token: "
							content={app.token}
							canCopy
							onRefresh={ () => { actions.regenToken(index) } }
							/>
					</Column>
				</Row>
				
				<Container styling='transparent' noPadding>
					<Column 
						width='1-1' 
						className="settings-collapse"
						noPadding
					>
						<Collapse expandIcon={expandIcon}>
							<Panel header={`OAuth Settings`} showArrow>
							<Container styling='transparent' noPadding>
								<Column
									width='1-1' 
									className='settings-section'
								>
									<TextView text={`OAuth Credentials: `}
										heading={4}
										align={`left`} 
									/>
									<Field
										title="Client ID: "
										content={app.oauth.client_id}
										canCopy
									/>
									<Field
										title="Client Secret: "
										content={app.oauth.client_secret}
										canCopy
									/>
									<Field
										title="Callback Url: "
										content={app.oauth.callback_url == '' ? 'https://' : app.oauth.callback_url}
										canCopy
										onSave={(value) => { actions.saveOAuthCallback(index, value) }}
									/>
								</Column>
							</Container>
							<Container styling='transparent' noPadding>
								<Column
									width='1-1' 
									className='settings-section'
								>
									<TextView text={`OAuth Scopes: `}
										heading={4}
										align={`left`} 
									/>
									{app.oauth.scopes.map( (scope, scope_index) => 
										<CheckBox 
											text={scope.description}
											isChecked={scope.enabled}
											onClick={(value) => { actions.setScope(index, scope_index, value) } }
											style={{ float: `left`, margin: `12px 10px 0 10px` }}
										/>
									)}
								</Column>
							</Container>
							</Panel>
							<Panel header={`Webhook Settings`} showArrow>
							<Container styling='transparent' noPadding>
								<Column
									width='1-1' 
									className='settings-section'
								>
									<Field
										title="Verification Secret: "
										content={app.webhook.verification_secret}
										canCopy
										onRefresh={() => { actions.regenVerificationSecret(index) }}
									/>
									<Field
										title="Webhook URL: "
										content={app.webhook.url=='' ? 'https://' : app.webhook.url}
										onSave={(value) => { actions.webhook.saveURL(index, value) } }
									/>
									<Field
										title="'siteid' (optional):"
										content={app.webhook.siteid}
										onSave={(value) => { actions.webhook.saveSiteID(index, value) }}
									/>
									<Field
										title="'roomid' (optional):"
										content={app.webhook.roomid}
										onSave={(value) => { actions.webhook.saveRoomID(index, value) }}
									/>
									<Field
										title="Contact (optional):"
										content={app.webhook.contact}
										onSave={(value) => { actions.webhook.saveContact(index, value) }}
									/>
								</Column>
							</Container>
							</Panel>
						</Collapse>
					</Column>
				</Container>
				<Container styling='transparent' noPadding>
					<Row width="1-1">
						<Column 
							width="1-1"
							className="bottom-app-column"
							keepInline
						>
							<Column 
								width="165px" 
								alignItems="column"
								className="dates-holder default tablet" 
								keepInline
							>
								<TextView text={`Created: ` + created + ` ago`} heading={5}
									align="left" style={styles.dates} color="white" />
								<TextView text={`Updated: ` + updated + ` ago`} heading={5}
									align="left" style={styles.dates} color="white" />
							</Column>

							<ButtonView
								text={`Delete`} 
								type={`remove`} 
								onClick={() => { actions.deleteConfirm(index) }} 
								fakeLink 
								style={{ cursor: `pointer` }} 
							/>
						</Column>
					</Row>
				</Container>
			</Container>
        </Panel>
        </Collapse>
		)
	}

	timeSince = (date) => {
	    const seconds = Math.floor((new Date() - date) / 1000)

	    let interval = Math.floor(seconds / 31536000)

	    if (interval > 1) {
	      return interval + ` years`
	    }
	    interval = Math.floor(seconds / 2592000)
	    if (interval > 1) {
	      return interval + ` months`
	    }
	    interval = Math.floor(seconds / 86400)
	    if (interval > 1) {
	      return interval + ` days`
	    }
	    interval = Math.floor(seconds / 3600)
	    if (interval > 1) {
	      return interval + ` hours`
	    }
	    interval = Math.floor(seconds / 60)
	    if (interval > 1) {
	      return interval + ` minutes`
	    }
	    return Math.floor(seconds) + ` seconds`
	}
}