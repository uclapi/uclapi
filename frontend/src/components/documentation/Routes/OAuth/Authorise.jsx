import propTypes from 'prop-types'
import React from 'react'

import Cell from './../../Cell.jsx'
import Table from './../../Table.jsx'
import Topic from './../../Topic.jsx'


const codeExamples = {
  python: `import requests

params = {
  "client_id": "123.456",
  "state": "1",
}
r = requests.get("https://uclapi.com/oauth/authorise", params=params)
print(r.json())`,

  shell: `curl -G https://uclapi.com/oauth/authorise/ \
-d client_id=123.456 \
-d state=1`,

  javascript: `fetch("https://uclapi.com/oauth/authorise/?client_id=123.456&state=1")
.then((response) => {
  return response.json()
})
.then((json) => {
  console.log(json);
})`,
}


const Authorise = ({ activeLanguage }) => (
  <div>
    <Topic
      activeLanguage={activeLanguage}
      codeExamples={codeExamples}
    >
      <h1 id="oauth/authorise">Authorise</h1>
      <p>
        Endpoint: <code>https://uclapi.com/oauth/authorise</code>
      </p>

      <Table
        name="Query Parameters"
      >
        <Cell
          name="client_id"
          requirement="required"
          example="123.456"
          description="Client ID of the authenticating app."
        />
        <Cell
          name="state"
          requirement="required"
          example="1"
          description="OAuth state."
        />
      </Table>
    </Topic>

    <Topic
      noExamples
    >
      <h2>Response</h2>
      <p>
        Redirection to authorise page.
            </p>
    </Topic>

    <Topic
      noExamples
    >
      <Table
        name="Errors"
      >
        <Cell
          name="Incorrect parameters supplied."
          description={
            `Gets returned when you have not supplied` +
            `a client_id and a state in your request.`
          }
        />
        <Cell
          name="App does not exist for client id."
          description="Gets returned when you supply an invalid client_id."
        />
        <Cell
          name="No callback URL set for this app."
          description={
            `Gets returned when you have`
            + `not set a callback URL for your app.`
          }
        />
        <Cell
          name="UCL has sent incomplete headers"
          description={
            `Gets returned when UCL sends us incomplete headers. `
            + `If the issues persist please contact the UCL API Team.`
          }
        />
      </Table>
    </Topic>
  </div>
)

export default Authorise

Authorise.propTypes = {
  activeLanguage: propTypes.string,
}

Authorise.defaultProps = {
  activeLanguage: `python`,
}

