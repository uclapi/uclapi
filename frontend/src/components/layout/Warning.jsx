import warning from 'Images/warning/warning.svg'
import {
  Button,
  Column,
  Container,
  Footer,
  ImageView,
  NavBar,
  Row,
  TextView,
} from 'Layout/Items.jsx'
import React from 'react'
import 'Styles/common/uclapi.scss'
import 'Styles/navbar.scss'

const Warning = ({ title, content }) => (
  <>
    <NavBar isScroll={false} />

    {/* Warning message */}

    <Container styling="splash-parallax" height="600px">
      <Row 
        width="2-3"
        horizontalAlignment="center"
        verticalAlignment="center"
      >
        
        <Column width="1-2" className="default">
          <ImageView
            src={warning}
            width="367px"
            height="405px"
            description="large exclamation point, warning!"
          />
        </Column>

        <Column width="1-2" alignItems="column">
          <TextView 
            heading="1"
            text={title}
            align="left"
          />
          {typeof content === `string`
            ? (
              <TextView 
                heading="3"
                text={content}
                align="left"
              />
            )
            : content}
          <Button link={`/`} centred>
            Home
          </Button>
        </Column>
      </Row>
    </Container>
    <Footer />
  </>
)

export default Warning