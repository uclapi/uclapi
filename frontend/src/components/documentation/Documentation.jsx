// UCL API navbar
import 'Styles/navbar.scss'

import React from 'react'

import { NavBar } from 'Layout/Items.jsx'

import GetInvolved from './GetInvolved/GetInvolved.jsx'
import Welcome from './Meta/Welcome.jsx'
import Authorise from './Routes/OAuth/Authorise.jsx'
import OAuthIntro from './Routes/OAuth/OAuthIntro.jsx'
import StudentNumber from './Routes/OAuth/StudentNumber.jsx'
import Token from './Routes/OAuth/Token.jsx'
import UserData from './Routes/OAuth/UserData.jsx'
import DesktopAvailability from './Routes/Resources/DesktopAvailability.jsx'
import GetBookings from './Routes/RoomBookings/GetBookings.jsx'
import GetEquiment from './Routes/RoomBookings/GetEquipment.jsx'
import GetFreeRooms from './Routes/RoomBookings/GetFreeRooms.jsx'
import GetRooms from './Routes/RoomBookings/GetRooms.jsx'
import Webhooks from './Routes/RoomBookings/Webhooks.jsx'
import GetPeople from './Routes/Search/GetPeople.jsx'
import GetDataCourses from './Routes/Timetable/GetDataCourses.jsx'
import GetDataCoursesModules from './Routes/Timetable/GetDataCoursesModules.jsx'
import GetDataDepartments from './Routes/Timetable/GetDataDepartments.jsx'
import GetDataModules from './Routes/Timetable/GetDataModules.jsx'
import GetPersonalTimetable from './Routes/Timetable/GetPersonalTimetable.jsx'
import GetTimetableByModules from './Routes/Timetable/GetTimetableByModules.jsx'
import WorkspacesGetImage from './Routes/Workspaces/GetImage.jsx'
import WorkspacesGetLastSensorUpdate from './Routes/Workspaces/GetLastSensorUpdate.jsx'
import WorkspacesGetLiveImage from './Routes/Workspaces/GetLiveImage.jsx'
import WorkspacesGetHistoricalTimeData from './Routes/Workspaces/GetSensorHistoricalTimeData.jsx'
import WorkspacesGetSensors from './Routes/Workspaces/GetSensors.jsx'
import WorkspaceGetSensorsSummary from './Routes/Workspaces/GetSensorSummary.jsx'
import WorkspacesGetSurveys from './Routes/Workspaces/GetSurveys.jsx'
import SectionHeader from './SectionHeader.jsx'
import Sidebar from './Sidebar.jsx'

export default class DocumentationComponent extends React.Component {

  render() {
    return (
      <div>
        <NavBar isScroll={false} />
        <Sidebar />
        <div className="main">
          <Welcome key={`Welcome`} />

          <SectionHeader
            key={`SectionHeaderWelcome`}
            link="oauth"
            title="OAuth"
          />
          <OAuthIntro key={`OAuthIntro`} />
          <Authorise key={`Authorise`} />
          <Token key={`Token`} />
          <UserData key={`UserData`} />
          <StudentNumber key={`StudentNumber`} />

          <SectionHeader
            link="roombookings"
            title="Room Bookings"
            key={`SectionHeaderRoomBookings`}
          />
          <GetRooms key={`GetRooms`} />
          <GetBookings key={`GetBookings`} />
          <GetEquiment key={`GetEquiment`} />
          <GetFreeRooms key={`GetFreeRooms`} />
          <Webhooks key={`Webhooks`} />

          <SectionHeader
            link="search"
            title="Search"
            key={`SectionHeaderSearch`}
          />
          <GetPeople key={`GetPeople`} />

          <SectionHeader
            link="timetable"
            title="Timetable"
            key={`SectionHeaderTimetable`}
          />
          <GetPersonalTimetable key={`GetPersonalTimetable`} />
          <GetTimetableByModules key={`GetTimetableByModules`} />
          <GetDataDepartments key={`GetDataDepartments`} />
          <GetDataModules key={`GetDataModules`} />
          <GetDataCourses key={`GetDataCourses`} />
          <GetDataCoursesModules key={`GetDataCoursesModules`} />

          <SectionHeader
            link="resources"
            title="Resources"
            key={`SectionHeaderResources`}
          />
          <DesktopAvailability key={`DesktopAvailability`} />

          <SectionHeader
            link="workspaces"
            title="Workspaces"
            key={`SectionHeaderWorkspaces`}
          />
          <WorkspacesGetSurveys key={`WorkspacesGetSurveys`} />
          <WorkspacesGetSensors key={`WorkspacesGetSensors`} />
          <WorkspacesGetHistoricalTimeData
            key={`WorkspacesGetHistoricalTimeData`}
          />
          <WorkspacesGetLastSensorUpdate
            key={`WorkspacesGetLastSensorUpdate`}
          />
          <WorkspaceGetSensorsSummary key={`WorkspaceGetSensorsSummary`} />
          <WorkspacesGetImage key={`WorkspacesGetImage`} />
          <WorkspacesGetLiveImage key={`WorkspacesGetLiveImage`} />

          <GetInvolved key={`GetInvolved`} />
        </div>
      </div>
    )
  }

}
