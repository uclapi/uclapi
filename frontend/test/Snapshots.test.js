import React from 'react'
import renderer from 'react-test-renderer'
import LogInLayout from '../src/components/appsettings/LogInLayout'
import AccordionIcon from "../src/components/dashboard/AccordionIcon"
import App from "../src/components/dashboard/App"
import SettingsLayout from "../src/components/appsettings/SettingsLayout"
import Navbar from "../src/components/dashboard/navbar"
import Authorise from "../src/pages/Authorise/Authorise"
import Dashboard from "../src/pages/Dashboard/Dashboard"
import HomePage from "../src/pages/Home/HomePage"
import Marketplace from "../src/pages/Marketplace/Marketplace"
import NotFoundPage from "../src/pages/404/404"
import InternalServerErrorPage from "../src/pages/500/500"
import AboutPage from "../src/pages/AboutPage/AboutPage"
import AppSettings from "../src/pages/AppSettings/AppSettings"
import Documentation from "../src/pages/Documentation/Documentation"
import Warning from "../src/pages/Warning/Warning"

describe(`Components`, () => {

  describe(`Appsettings`, () => {
    it(`LogInLayout`, () => {
      const tree = renderer.create(<LogInLayout url="https://www.uclapi.com"/>).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it(`SettingsLayout`, () => {
      const tree = renderer.create(
        <SettingsLayout authorised_apps={[`app1`]}/>,
      ).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  describe(`Dashboard`, () => {
    it(`AccordionIcon`, () => {
      const tree = renderer.create(
        <AccordionIcon isActive/>,
      ).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it(`App`, () => {
      const tree = renderer.create(
        <App app={{
          "name": `sample-project`,
          "id": `sample-log`,
          "token": `uclapi-token`,
          "created": `2020-10-21T11:15:46.932`,
          "updated": `2020-10-21T11:16:38.648`,
          "oauth": {
            "client_id": `123.456`,
            "client_secret": `secrets`,
            "callback_url": ``,
            "scopes": [
              {
                "name": `timetable`,
                "description": `Personal Timetable`,
                "enabled": true,
              },
              {
                "name": `student_number`,
                "description": `Student Number`,
                "enabled": false,
              },
            ],
          },
          "webhook": {
            "verification_secret": `secrets`,
            "url": ``,
            "siteid": ``,
            "roomid": ``,
            "contact": ``,
          },
          "analytics": {
            "requests": 392,
            "remaining_quota": 10000,
            "users": 0,
            "users_per_dept": [],
          },
        }}
        />,
      ).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it(`Navbar`, () => {
      const tree = renderer.create(
        <Navbar/>,
      ).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})

describe(`Pages`, () => {
  const pages = {
    "Authorise": [[`Authorise`, Authorise]],
    "Dashboard": [[`Dashboard`, Dashboard]],
    "Home": [[`HomePage`, HomePage]],
    "Marketplace": [[`Marketplace`, Marketplace]],
    "404": [[`NotFoundPage`, NotFoundPage]],
    "505": [[`InternalServerErrorPage`, InternalServerErrorPage]],
    "AboutPage": [[`AboutPage`, AboutPage]],
    "AppSettings": [[`AppSettings`, AppSettings]],
    "Documentation": [[`Documentation`, Documentation]],
    "Warning": [[`Warning`, Warning]],
  }

  for (const page in pages) {
    describe(page, () => {
      // eslint-disable-next-line sonarjs/no-identical-functions
      it.each(pages[page])(`%s`, (name, component) => {
        const tree = renderer.create(
          <component/>,
        ).toJSON()
        expect(tree).toMatchSnapshot()
      })
    })
  }
})
