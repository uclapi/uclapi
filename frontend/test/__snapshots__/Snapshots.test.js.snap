// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Appsettings LogInLayout 1`] = `
<div
  className="log-in-layout"
>
  <div
    className="container"
  >
    <h1>
      Sign in to manage app settings
    </h1>
    <a
      href="https://www.uclapi.com"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text"
        disabled={false}
        label="Log in"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        />
      </button>
    </a>
  </div>
</div>
`;

exports[`Components Appsettings SettingsLayout 1`] = `
<div
  className="container splash-parallax vertical-padding undefined"
  style={
    Object {
      "height": "fit-content",
      "minHeight": "100vh",
    }
  }
>
  <div
    className="no-vertical-align"
  >
    <div
      className="row undefined"
      style={
        Object {
          "justifyContent": "center",
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": "50px",
          "width": "90%",
        }
      }
    >
      <h1
        style={
          Object {
            "margin": "20px 0",
            "marginLeft": 0,
            "textAlign": "left",
          }
        }
      >
        Account
      </h1>
      <div
        className="column "
        style={
          Object {
            "margin": 0,
            "padding": 0,
            "width": "100%",
          }
        }
      >
        <div
          className="uclapi-card uclapi-card-default uclapi-card-shadow"
          style={
            Object {
              "padding": 0,
            }
          }
        >
          <div
            className="container transparent undefined"
            style={Object {}}
          >
            <div
              className="no-vertical-align"
            >
              <div
                className="row undefined"
                style={
                  Object {
                    "justifyContent": "center",
                    "marginLeft": "auto",
                    "marginRight": "auto",
                    "width": "100%",
                  }
                }
              >
                <div
                  className="column-always-inline "
                  style={
                    Object {
                      "margin": 0,
                      "padding": 0,
                      "width": "60%",
                    }
                  }
                >
                  <div
                    className="uclapi-card uclapi-card-transparent"
                    style={
                      Object {
                        "padding": "30px",
                      }
                    }
                  >
                    <h2
                      style={
                        Object {
                          "marginBottom": 0,
                          "marginLeft": 0,
                          "textAlign": "left",
                        }
                      }
                    >
                      
                    </h2>
                    <h4
                      style={
                        Object {
                          "marginBottom": 0,
                          "marginLeft": 0,
                          "textAlign": "left",
                        }
                      }
                    >
                      
                    </h4>
                  </div>
                </div>
                <div
                  className="column-always-inline "
                  style={
                    Object {
                      "margin": 0,
                      "padding": 0,
                      "width": "40%",
                    }
                  }
                >
                  <div
                    className="uclapi-card uclapi-card-transparent"
                    style={
                      Object {
                        "padding": "30px",
                      }
                    }
                  >
                    <a
                      className="uclapi-button default-transition background-color-transition alternate-button "
                      href="/logout"
                      style={
                        Object {
                          "cursor": "pointer",
                          "float": "right",
                          "width": "fit-content",
                        }
                      }
                    >
                      Logout
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <h1
        style={
          Object {
            "margin": "20px 0",
            "marginLeft": 0,
            "textAlign": "left",
          }
        }
      >
        Permissions
      </h1>
      <div
        className="column "
        style={
          Object {
            "margin": 0,
            "padding": 0,
            "width": "100%",
          }
        }
      >
        <div
          className="uclapi-card uclapi-card-default uclapi-card-shadow"
          style={
            Object {
              "padding": 0,
            }
          }
        >
          <div
            className="container transparent undefined"
            style={Object {}}
          >
            <div
              className="no-vertical-align"
            >
              <div
                className="row undefined"
                style={
                  Object {
                    "justifyContent": "center",
                    "marginLeft": "auto",
                    "marginRight": "auto",
                    "paddingBottom": 20,
                    "paddingTop": 20,
                    "width": "100%",
                  }
                }
              >
                <div
                  className="column default"
                  style={
                    Object {
                      "width": "50%",
                    }
                  }
                >
                  <div
                    className="image-view"
                    style={
                      Object {
                        "height": "405px",
                        "marginLeft": "auto",
                        "marginRight": "auto",
                        "width": "367px",
                      }
                    }
                  >
                    <img
                      alt="ucl asssitant screen shot"
                      height="405px"
                      src="test-file-stub"
                      style={Object {}}
                      width="367px"
                    />
                  </div>
                </div>
                <div
                  className="column "
                  style={
                    Object {
                      "flexDirection": "column",
                      "width": "50%",
                    }
                  }
                >
                  <h2
                    style={
                      Object {
                        "textAlign": "left",
                      }
                    }
                  >
                    No authorised apps yet. Try UCL Assistant?
                  </h2>
                  <h5
                    style={
                      Object {
                        "textAlign": "left",
                      }
                    }
                  >
                    UCL Assistant is a productivity app for students and staff at UCL. It has been designed by students to be as reliable and user friendly as possible.
                  </h5>
                  <h5
                    style={
                      Object {
                        "textAlign": "left",
                      }
                    }
                  >
                    The app has many different features but one of our favourites is the ability to view your timetable in a weekly view. The app highlights your current commitments to save you time as you rush about University. Check it out below!
                  </h5>
                  <a
                    className="uclapi-button default-transition background-color-transition alternate-button "
                    href="/marketplace/uclassistant"
                    style={
                      Object {
                        "marginLeft": "auto",
                        "marginRight": "auto",
                        "width": "fit-content",
                      }
                    }
                  >
                    Download
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Components Dashboard AccordionIcon 1`] = `
<div
  className="accordion-icon"
>
  <svg
    fill="none"
    height="1em"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <polyline
      points="6 9 12 15 18 9"
    />
  </svg>
</div>
`;

exports[`Components Dashboard App 1`] = `
<div
  className="rc-collapse"
  role={null}
>
  <div
    className="rc-collapse-item"
  >
    <div
      aria-expanded={false}
      className="rc-collapse-header"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="accordion-icon"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="9 18 15 12 9 6"
          />
        </svg>
      </div>
      sample-project
    </div>
  </div>
</div>
`;

exports[`Components Dashboard Navbar 1`] = `
<div
  className="navbar centered"
>
  <img
    src="test-file-stub"
  />
  <div
    className="logo-text"
  >
    <div>
      UCL API
    </div>
  </div>
</div>
`;

exports[`Pages 404 NotFoundPage 1`] = `<component />`;

exports[`Pages 505 InternalServerErrorPage 1`] = `<component />`;

exports[`Pages AboutPage AboutPage 1`] = `<component />`;

exports[`Pages AppSettings AppSettings 1`] = `<component />`;

exports[`Pages Authorise Authorise 1`] = `<component />`;

exports[`Pages Dashboard Dashboard 1`] = `<component />`;

exports[`Pages Documentation Documentation 1`] = `<component />`;

exports[`Pages Home HomePage 1`] = `<component />`;

exports[`Pages Marketplace Marketplace 1`] = `<component />`;

exports[`Pages Warning Warning 1`] = `<component />`;
